test/e2e/operator_test.go

```diff
@@ -1,26 +1,316 @@
 package e2e
 
 import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
 	"testing"
+	"time"
 
+	corev1 "k8s.io/api/core/v1"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	kapierrs "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/kubernetes"
+	policyclientv1beta1 "k8s.io/client-go/kubernetes/typed/policy/v1beta1"
 
 	"github.com/openshift/cluster-kube-controller-manager-operator/pkg/operator/operatorclient"
 	test "github.com/openshift/cluster-kube-controller-manager-operator/test/library"
+	"github.com/prometheus/common/model"
 )
 
-func TestOperatorNamespace(t *testing.T) {
+const (
+	kcmoNamespace        = "openshift-kube-controller-manager-operator"
+	kcmoSigningKeySecret = "kube-controller-manager-operator-serving-cert"
+)
+
+func TestE2E(t *testing.T) {
```

Comment:
I'm not sure this nesting is needed. All tests under `test/e2e` are e2e tests, and for example , `TestSATokenSignerControllerSyncCerts` is in another file and won't be in this hierarchy.

Reply:
Nesting tests allows setup in the enclosing function to be shared, but I would agree that unless the setup is expensive (e.g. creating resources) I'm not sure it makes sense to use nesting vs creating a helper that tests can call.

Reply:
@sanchezl Ah, and to your point, naming this `TestE2E` probably doesn't make sense unless this is the entrypoint for running all e2e tests rather than just some of them.

Reply:
updated, took out the nesting.